### PR TITLE
Test mode

### PR DESCRIPTION
**Pull request by asiniy**
It does what it declares!

Look at `README.md` documentation to figure out what's going on.
If everything looks nice => release a new version please

### CODE REVIEW

**Comment by asiniy** [comment]
Dear maintainers, is anyone here?
/cc @scrogson

**Comment by doomspork** [comment]
Howdy @asiniy!  Apologizes for the delay, I've been away for some time and am catching up on dozens of PRs 😰 

A test mode would be _terrific_ and is something others have expressed an interest in.  In fact this was recently brought up by @Papipo in #70.  @Papipo I would love to get your thoughts on this implementation in addition to the @ueberauth/developers team, would this satisfy your requirements?

**Comment by Papipo** [comment]
I think this is probably a good start.
Some things I miss, but that can be implemented in the future (even by me if I find the time):

- Adding mocks at runtime.
- Forcing an authentication failure.
- Something else I don't remember now.

Thanks a lot @asiniy!

**Comment by yordis** [comment]
I would prefer to create some helper functions on the testing instead of adding this config if posible or also, what about changing the module itself?

Instead of `Ueberauth.Strategy.Twitter` you would use a mock module.

I think we are tackling this the way that Ruby did it but I believe we should go with more adpater based architecture for this (just swap the implementation of the strategy)

File: README.md [hunk 1]
```diff
@@ -176,6 +176,38 @@ providers: [
 ]
 ```
 
+## Test Mode
+
+UeberAuth has some facilities for mocking its authentication flow when you are performing integration tests. Set up UeberAuth with `test_mode` option:
```
Comment (scrogson) [inline README.md]: `Ueberauth`

**Comment by yordis** [comment]
The PR is a bit old, so I am gonna close it, feel free to reopen it and continue the work. That being said, I will recommend writing a `TestStrategy` as an alternative.

**Pull request closed without merging**